test/integration/targets/get_certificate/tests/validate.yml

```diff
@@ -0,0 +1,90 @@
+---
+- name: Test get_certificate
+  connection: local
+  hosts: localhost
+  gather_facts: false
+  tasks:
+
+    - name: Get servers certificate
+      get_certificate:
+        host: "{{ httpbin_host }}"
+        port: 443
+      register: result
+
+    - debug: var=result
+
+    - assert:
+        that:
+          # This module should never change anything
+          - "result.changed == false"
+          - "result.failed == false"
+          # We got the correct ST from the cert
+          - "'North Carolina' == result.cert.ST"
+
+    - name: Connect to http port (will fail because there is no SSL cert to get)
+      get_certificate:
+        host: "{{ httpbin_host }}"
+        port: 80
+      register: result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - "result.changed == false"
+          - "result.failed == true"
```

Comment:
These should be `result is not changed` and `result is failed` (similar for the ones below).

Reply:
Updated and pushed.